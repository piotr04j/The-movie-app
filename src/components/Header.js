import React from 'react';

export default () => <h2>Header</h2>;
