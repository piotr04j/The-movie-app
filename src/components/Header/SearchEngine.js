import React, { Component } from 'react'
import { connect } from 'react-redux';
import { fetchSearchedData } from '../../store/actions';

export class SearchEngine extends Component {

    state = {
        formValue: ''
    }

    handleInputValue = (e) =>{
        this.setState({formValue: e.target.value})
    }

    handleSubmit = (e) => {
        e.preventDefault()
        this.props.fetchSearchedData(this.state.formValue);
        this.setState({formValue: ''});
    }

    render () {
        return (
            <form className="form-inline" onSubmit={this.handleSubmit}>
                <input className="form-control mr-sm-2" type="text" placeholder="Search..." value={this.state.formValue} aria-label="Search" onChange={this.handleInputValue} />
                <input className="btn btn-outline-success my-2 my-sm-0" type="submit" value="Search" />
            </form>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchSearchedData: (data) => dispatch(fetchSearchedData(data))
    }
}

export default connect(null,mapDispatchToProps)(SearchEngine);

