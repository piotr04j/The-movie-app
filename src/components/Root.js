import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import reducer from '../store/reducer';
import { composeWithDevTools } from 'redux-devtools-extension';

const RootStore = ({children, initialState = { err: null, searchedData: null }}) => {
    const store = createStore(reducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return (
        <Provider store={store}>
            {children}
        </Provider>
    )
};

export default RootStore;