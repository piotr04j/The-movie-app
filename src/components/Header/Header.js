import React from 'react';
import { NavLink } from 'react-router-dom';
import SearchEngine from './SearchEngine';


const Header = () => {

    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
            <div className="container">
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav mr-auto">
                        <li className="nav-item active">
                            <NavLink activeStyle={{fontWeight: "bold"}} className="nav-link" exact to="/">Home<span className="sr-only">(current)</span></NavLink>
                        </li>
                        <li className="nav-item active">
                            <NavLink activeStyle={{fontWeight: "bold"}} className="nav-link" to="/movies-list/1">Movies<span className="sr-only">(current)</span></NavLink>
                        </li>
                        <li className="nav-item active">
                            <NavLink activeStyle={{fontWeight: "bold"}} className="nav-link" to="/tv-list/1">TV Shows<span className="sr-only">(current)</span></NavLink>
                        </li>
                    </ul>
                    <span className="navbar-text">
                        <SearchEngine />
                    </span>
                </div>
            </div>
        </nav>
    )
}


export default Header;
