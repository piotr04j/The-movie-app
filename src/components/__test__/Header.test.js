import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Header  from '../Header/Header';
import Root from  '../Root';

it('renders correctly', () => {
    const tree = renderer.create(
        <Root>
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        </Root>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

