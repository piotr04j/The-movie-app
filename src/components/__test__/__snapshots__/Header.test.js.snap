// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary mb-3"
>
  <div
    className="container"
  >
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Home
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item active"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/movies-list/1"
            onClick={[Function]}
          >
            Movies
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item active"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/tv-list/1"
            onClick={[Function]}
          >
            TV Shows
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
      </ul>
      <span
        className="navbar-text"
      >
        <form
          className="form-inline"
          onSubmit={[Function]}
        >
          <input
            aria-label="Search"
            className="form-control mr-sm-2"
            onChange={[Function]}
            placeholder="Search..."
            type="text"
            value=""
          />
          <input
            className="btn btn-outline-success my-2 my-sm-0"
            type="submit"
            value="Search"
          />
        </form>
      </span>
    </div>
  </div>
</nav>
`;
